### ISSUE TITLE
use XDG baasedir spec instead of ~/.dotfile

### ISSUE THREAD

**Issue by flying-sheep** [oakes/SolidOak#38]
[`xdg_basedir`](http://kiran-kp.github.io/xdg-basedir/xdg_basedir/index.html) has [`get_config_home()`](http://kiran-kp.github.io/xdg-basedir/xdg_basedir/dirs/fn.get_config_home.html) which you can use to get the correct directory to put config files into on linux.

on windows, it should go into `%appdata%`.

for backwards compatibility, it should first try to read from the correct location, then from the old `~/.solidoakrc`, and use the first one it can find. if it can’t find anything, it should create the config file in the correct location.

**Comment by oakes** [comment]
Generally I'd do this, but the reason I didn't in this case was to be consistent with Vim, which also does not use that spec. Experienced Vim users would expect it to be in the root of the home dir.

**Comment by flying-sheep** [comment]
simply allow both then. if the config file in the correct location cannot be found, fall back to ~/.soakrc

**Comment by oakes** [comment]
Yeah I can do that (actually, that's what I do in my other IDE, Nightcode).

**Comment by flying-sheep** [comment]
:+1: great, thanks!
